vmbindings/dummyvm/src/tests/allocator_info.rs

```diff
@@ -0,0 +1,56 @@
+// GITHUB-CI: MMTK_PLAN=all
+
+use mmtk::util::alloc::AllocatorInfo;
+use mmtk::util::options::PlanSelector;
+use mmtk::AllocationSemantics;
+
+use crate::test_fixtures::{Fixture, MMTKSingleton};
+use crate::DummyVM;
+
+lazy_static! {
+    static ref MMTK_SINGLETON: Fixture<MMTKSingleton> = Fixture::new();
+}
+
+#[test]
+fn test_allocator_info() {
+    MMTK_SINGLETON.with_fixture(|fixture| {
+        let selector = mmtk::memory_manager::get_allocator_mapping(
+            &fixture.mmtk,
+            AllocationSemantics::Default,
+        );
+        let base_offset = mmtk::plan::Mutator::<DummyVM>::get_allocator_base_offset(selector);
+        let allocator_info = AllocatorInfo::new::<DummyVM>(selector);
+
+        match *fixture.mmtk.get_options().plan {
+            PlanSelector::NoGC
+            | PlanSelector::Immix
+            | PlanSelector::SemiSpace
+            | PlanSelector::GenCopy
+            | PlanSelector::GenImmix
+            | PlanSelector::MarkCompact
+            | PlanSelector::StickyImmix => {
+                match allocator_info {
+                    // These plans all use bump pointer allocator. The first field at base offset is tls, and the second field is the BumpPointer struct.
+                    AllocatorInfo::BumpPointer {
+                        bump_pointer_offset,
+                    } => assert_eq!(
+                        base_offset + mmtk::util::constants::BYTES_IN_ADDRESS,
+                        bump_pointer_offset
+                    ),
+                    _ => panic!("Expected AllocatorInfo for a bump pointer allocator"),
+                }
```

Comment:
The `match` statement looks OK, but since Rust 1.65, we can also use the `let-else` statement when we expect it to match exactly one case or diverge (return, break, continue or panic) if not matching.  By doing this, we can reduce the level of nesting.

```suggestion
                // These plans all use bump pointer allocator.
                let AllocatorInfo::BumpPointer {
                    bump_pointer_offset,
                } = allocator_info
                else {
                    panic!("Expected AllocatorInfo for a bump pointer allocator");
                };

                // In all of those plans, the first field at base offset is tls, and the second field is the BumpPointer struct.
                assert_eq!(
                    base_offset + mmtk::util::constants::BYTES_IN_ADDRESS,
                    bump_pointer_offset
                );
```

Reply:
Done